### PR TITLE
Add System.IO.Hashing ref csproj

### LINKED ISSUE #77846: System.IO.Hashing's ref is missing a csproj
As far as I can tell, it's the only ref folder in all of libraries that lacks one.  Whether or not we're publishing a ref assembly in the nuget package, it means the System.IO.Hashing.cs in that folder isn't being validated.

cc: @ViktorHofer, @bartonjs

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/runtime/issues/77846

### CODE REVIEW

**Comment by vcsjones** [comment]
Most (all? I just did a few spot checks) things include the /ref project in the solution file. Should this be added to System.IO.Hashing.sln?

**Review by bartonjs — APPROVED** [review]

File: eng/ApiCompatExcludeAttributes.txt [hunk 1]
```diff
@@ -2,3 +2,4 @@ T:System.Diagnostics.CodeAnalysis.MemberNotNullAttribute
 T:System.Diagnostics.DebuggerGuidedStepThroughAttribute
 T:System.Runtime.CompilerServices.EagerStaticClassConstructionAttribute
 T:System.Runtime.CompilerServices.NativeIntegerAttribute
+T:System.Runtime.CompilerServices.SkipLocalsInitAttribute
```
Comment (ViktorHofer) [inline eng/ApiCompatExcludeAttributes.txt:5]: Why wasn't this excluded before? Or to rephrase, what makes System.IO.Hashing special that it requires this attribute exclusion?

File: src/libraries/System.IO.Hashing/ref/System.IO.Hashing.csproj [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>$(NetCoreAppCurrent);$(NetCoreAppMinimum);netstandard2.0;$(NetFrameworkMinimum)</TargetFrameworks>
+    <NoWarn>$(NoWarn);CS0809</NoWarn>
```
Comment (ViktorHofer) [inline src/libraries/System.IO.Hashing/ref/System.IO.Hashing.csproj:4]: nit: add an xml comment to explain why this NoWarn is required.

**Review by ViktorHofer — COMMENTED** [review]
LGTM aside from two comments

File: eng/ApiCompatExcludeAttributes.txt (hunk 1 above)
Comment (stephentoub) [reply eng/ApiCompatExcludeAttributes.txt:5]: Everywhere else we use it at the module level.  This is the first time we've used it at the class level.

Comment (ViktorHofer) [reply eng/ApiCompatExcludeAttributes.txt:5]: Why can't we apply it for the entire System.IO.Hashing module?

Comment (stephentoub) [reply eng/ApiCompatExcludeAttributes.txt:5]: We could, it's just broader allocation than we've typically done outside of netcoreapp where this attribute is automatically added at to every module. Regardless, it makes sense to invite the attribute here; it should be in ref assemblies.

**Pull request merged**